File: app/src/main/java/com/engineeringforyou/basesite/presentation/searchsite/presenter/SearchSitePresenter.java
package com.engineeringforyou.basesite.presentation.searchsite.presenter;

import android.widget.EditText;

import com.engineeringforyou.basesite.presentation.searchsite.views.SearchSiteView;

public interface SearchSitePresenter {

    void bind(SearchSiteView view);

    void unbindView();

    void setupOperator();

    void searchSite(String search);

    void watchChanges(EditText view);

    void saveOperator(int operatorIndex);
}

File: app/src/main/java/com/engineeringforyou/basesite/presentation/searchsite/presenter/SearchSitePresenter.kt
package com.engineeringforyou.basesite.presentation.searchsite.presenter

import android.widget.EditText

import com.engineeringforyou.basesite.presentation.searchsite.views.SearchSiteView

interface SearchSitePresenter {

    fun bind(view: SearchSiteView)

    fun unbindView()

    fun setupOperator()

    fun searchSite(operatorIndex: Int, search: String)

    fun showMap(operatorIndex: Int)

    fun watchChanges(view: EditText)

    fun saveOperator(operatorIndex: Int)
}
